### ISSUE TITLE
Support Link Provider and Relative Paths 

### ISSUE THREAD

**Issue by arciisine** [microsoft/vscode#113293]
## TL;DR
I am currently trying to support links within a vscode plugin, that are relative to the current working directory of the terminal, similar to how the relative paths in the terminal work today.  On Unix based systems, I'm able to access `/proc/{pid}/cwd` as a symlink to the terminal's shell cwd, but this is not cross platform, and seems unreliable.

I'd love to see a consistent way to expose the `cwd` of the terminal, or provide a way for the link provider to resolve relative paths in the context of the terminal it was clicked in.

## Detailed Example
I'm currently using `vscode.window.registerTerminalLinkProvider` to great effect.  

One of the use cases I'm currently supporting is an internalized categorization that allows for direct mapping to a specific file/class. The purpose of this is to allow for a more compact log output.

```bash
2020-12-22T14:28:58.500Z debug [@trv:base/shutdown:48] Starting { name: '@trv:model-mongo/service' }
```
The token `@trv:base/shutdown:48` is a shorthand that would expand to a `node_modules` reference.  

The problem arises when I need to deal with a path that is relative to the terminals's current working directory (cwd), in a similar fashion to how the relative paths work in the terminal today.

```bash
2020-12-22T14:28:58.500Z debug [src/service:48] Request received
```
The token `src/service:48` is a shorthand that should expand to a `{cwd}/src/service.ts`, but I have no mechanism to determine the working directory in a cross platform (or reliable) way.

**Comment by Tyriar** [comment]
You can see how we get the cwd for relative link support here:

https://github.com/microsoft/vscode/blob/7a34c6d622724575122465bb46078af06e07ee44/src/vs/workbench/contrib/terminal/node/terminalProcess.ts#L358-L406

Unfortunately on Windows it's impossible to do this, I confirmed this in an internal email thread with folks working on the Windows Kernel.

This could potentially be an API in the future, I'd recommend you can just copy how we do it for now. Be careful not to call lsof a lot, especially on macos big sur (https://github.com/microsoft/vscode/issues/105446). This is probably what such an API would look like:

```ts
export interface Terminal {
  cwd: Thenable<string>;
}
```
